lib/src/authenticate.tsx

```diff
@@ -111,6 +125,10 @@ const AuthProvider: FunctionComponent<PropsWithChildren<AuthProviderPropsInterfa
         return AuthClient.on(hook, callback);
     };
     const trySignInSilently = () => AuthClient.trySignInSilently(state, dispatch);
+    
+    const [ errors, setErrors ] = useState<AsgardeoAuthException[]>([]);
+
+    const addError = useCallback((error: AsgardeoAuthException) => setErrors([...errors, error]), []);
```

Comment:
Why do we need a callback here? Can't we call `setErrors` directly?
I think we can do something like this on L186:
```
setErrors(errors => [ ...errors, error ]);
```
And any reason why we append errors to an array?